xwiki-platform-core/xwiki-platform-flamingo/xwiki-platform-flamingo-skin/xwiki-platform-flamingo-skin-resources/src/main/resources/flamingo/flamingo.js

```diff
@@ -40,14 +40,87 @@ require(['jquery', 'bootstrap'], function($) {
         }
       }
     });
-
-    // When the drawer is close, collapse sub items
-    $(body).on('drawer.closed', function() {
-      $('.drawer-menu-sub-item').removeClass('in').attr('aria-expanded', 'false');
-    });
-
     // Activate the popover when hovering the Translate button.
     var translateButton = $('#tmTranslate [data-toggle="popover"]');
     translateButton.attr('title', translateButton.attr('data-title')).popover();
   });
 });
+
+/*
+  Handle the behavior of all drawers on the page.
+  In order to create a drawer that is compatible with this script, two elements are necessary:
+  * A drawer toggler button, that will have:
+   ** a reference to the drawer container in its attribute 'aria-control'
+   ** the class 'drawer-toggle'
+  * A drawer container, that will have:
+   ** a unique ID
+   ** the class 'drawer-nav' (mostly for style) 
+   ** the class 'closed' if the drawer is supposed to start in a closed state.
+   ** a label 
+   ** The content to display inside. It's asserted that at least one element in this content can receive focus.
+  For an example of drawer creation, see #tmDrawerActivator and #tmDrawer.
+ */
+require(['jquery'], function($) {
+  // The overlay is the same whatever the drawer opened.
+  const drawerOverlay = $(".drawer-overlay");
+  $('.drawer-toggle').each(function(index) {
+    // Setting up the drawer.
+    let drawerContainerToggler = $(this);
+    let drawerId = drawerContainerToggler.attr('aria-controls');
+    let drawerContainer = $('#'+drawerId);
+    let focusableElements = drawerContainer.find('button, a, input:not([type="hidden"]), ' +
+     'select, textarea, [tabindex]:not([tabindex="-1"])');
```

Comment:
Shouldn't you check here that these elements are actually visible and that they aren't disabled in some way?

Reply:
AFAIK there's no easy way to do this. We'd need to access the value of the css property 'display'. We could rule out all elements that have the 'hidden' HTML attribute on, but this makes the request more complex and AFAIK all elements shown in the drawer are enabled.